codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaderDateFormatter.java

```diff
@@ -0,0 +1,421 @@
+                    }
+                }
+            } else {
+                // invalid char
+                return false;
+            }
+        }
+
+        if (currentPartValue > 0) {
+            // pending seconds
+            localSeconds = currentPartValue;
+        }
+
+        if (localHours >= 0 && localMinutes >= 0 && localSeconds >= 0) {
+            hours = localHours;
+            minutes = localMinutes;
+            seconds = localSeconds;
+            return true;
+        }
+
+        return false;
+    }
+
+    private boolean tryParseDayOfMonth(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        if (len != 1 && len != 2) {
+            return false;
+        }
+
+        int localDayOfMonth = 0;
+        for (int i = tokenStart; i < tokenEnd; i++) {
+            char c = txt.charAt(i);
+            if (isDigit(c)) {
+                localDayOfMonth = localDayOfMonth * 10 + getNumericalValue(c);
+            } else {
+                // invalid
+                return false;
+            }
+        }
+
+        if (localDayOfMonth > 0) {
+            dayOfMonth = localDayOfMonth;
+            return true;
+        }
+
+        return false;
+    }
+
+    private boolean tryParseMonth(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        if (len == 3) {
+            String tokenString = txt.subSequence(tokenStart, tokenEnd).toString();
+
+            if ("Jan".equalsIgnoreCase(tokenString)) {
+                month = Calendar.JANUARY;
+            } else if ("Feb".equalsIgnoreCase(tokenString)) {
+                month = Calendar.FEBRUARY;
+            } else if ("Mar".equalsIgnoreCase(tokenString)) {
+                month = Calendar.MARCH;
+            } else if ("Apr".equalsIgnoreCase(tokenString)) {
+                month = Calendar.APRIL;
+            } else if ("May".equalsIgnoreCase(tokenString)) {
+                month = Calendar.MAY;
+            } else if ("Jun".equalsIgnoreCase(tokenString)) {
+                month = Calendar.JUNE;
+            } else if ("Jul".equalsIgnoreCase(tokenString)) {
+                month = Calendar.JULY;
+            } else if ("Aug".equalsIgnoreCase(tokenString)) {
+                month = Calendar.AUGUST;
+            } else if ("Sep".equalsIgnoreCase(tokenString)) {
+                month = Calendar.SEPTEMBER;
+            } else if ("Oct".equalsIgnoreCase(tokenString)) {
+                month = Calendar.OCTOBER;
+            } else if ("Nov".equalsIgnoreCase(tokenString)) {
+                month = Calendar.NOVEMBER;
+            } else if ("Dec".equalsIgnoreCase(tokenString)) {
+                month = Calendar.DECEMBER;
+            }
+        }
+
+        return month != -1;
+    }
+
+    private boolean tryParseYear(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        if (len != 2 && len != 4) {
```

Comment:
ok. If you don't mind, I'd rather use `if-else` than `switch`, as variable names are not scoped in `case` branches (and I'd rather not do smart things with falling through cases as it would hurt readability).